src/portable/synopsys/dwc2/dcd_dwc2.c

```diff
@@ -351,6 +382,12 @@ static void bus_reset(uint8_t rhport) {
   // Setup the control endpoint 0
   _allocated_fifo_words_tx = 16;
 
+  // DMA needs extra space for processing
+  if(dma_enabled(rhport)) {
+    uint16_t reserved = _dwc2_controller[rhport].ep_fifo_size / 4- dwc2->ghwcfg3_bm.total_fifo_size;
```

Comment:
Kernel use `dwc2->ghwcfg3_bm.total_fifo_size` it should be reliable. We can even remove `_dwc2_controller[rhport].ep_fifo_size` and use this field instead except GD32VF103 looks strange.

Reply:
yeah, this is the max epinfo where scatt/gather dma is used. We can use gdfifocfg to dynamic change the epinfo base to reduce the DMA region for simple DMA mode.